pkgs/development/interpreters/python/hooks/python-namespaces-hook.sh

```diff
@@ -17,16 +17,19 @@ pythonNamespacesHook() {
         for pathSegment in ${pathSegments[@]}; do
             constructedPath=${constructedPath}/${pathSegment}
             pathToRemove=${constructedPath}/__init__.py
-            pycachePath=${constructedPath}/__pycache__/__init__*
+            pycachePath=${constructedPath}/__pycache__/
 
             if [ -f "$pathToRemove" ]; then
                 echo "Removing $pathToRemove"
                 rm "$pathToRemove"
             fi
 
-            if [ -f "$pycachePath" ]; then
-                echo "Removing $pycachePath"
-                rm "$pycachePath"
+            if [ -d "$pycachePath" ]; then
+                # can be interpreter specific. E.g. __init__.cpython-38.pyc
+                for f in $(@findutils@/bin/find $pycachePath -name '__init__*'); do
```

Comment:
Do we need to worry about handling spaces in filenames here?

Reply:
I don't believe so, since nix wont introduce them, and python modules don't allow spaces

Reply:
but, if you know of a better way, them I'm all ears

Reply:
applied suggestion below, thanks